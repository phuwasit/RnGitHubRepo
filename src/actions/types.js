export const FETCH_REPOS_REQUEST = 'FETCH_REPOS_REQUEST'
export const FETCH_REPOS_SUCCESS = 'FETCH_REPOS_SUCCESS'
export const FETCH_REPOS_FAILURE = 'FETCH_REPOS_FAILURE'

export const SELECT_REPO = 'SELECT_REPO'

export const FETCH_CONTRIBUTORS_REQUEST = 'FETCH_CONTRIBUTORS_REQUEST'
export const FETCH_CONTRIBUTORS_SUCCESS = 'FETCH_CONTRIBUTORS_SUCCESS'
export const FETCH_CONTRIBUTORS_FAILURE = 'FETCH_CONTRIBUTORS_FAILURE'
