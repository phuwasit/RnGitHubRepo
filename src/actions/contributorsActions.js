import * as types from './types'

export const fetchContributors = (owner, repoName) => async (dispatch) => {
  const uri = `https://api.github.com/repos/${owner}/${repoName}/contributors`

  dispatch(fetchContributorsRequest())
  try {
    const response = await fetch(uri)
    const repos = await response.json()
    dispatch(fetchContributorsSuccess(repos))
  } catch (error) {
    dispatch(fetchContributorsFailure(error))
  }
}

export const fetchContributorsRequest = () => ({
  type: types.FETCH_CONTRIBUTORS_REQUEST,
})

export const fetchContributorsSuccess = data => ({
  type: types.FETCH_CONTRIBUTORS_SUCCESS,
  payload: data,
})

export const fetchContributorsFailure = error => ({
  type: types.FETCH_CONTRIBUTORS_FAILURE,
  payload: error,
})
