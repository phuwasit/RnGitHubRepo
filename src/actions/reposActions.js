import * as types from './types'

export const fetchRepos = username => async (dispatch) => {
  const uri = `https://api.github.com/users/${username}/repos`

  dispatch(fetchReposRequest())
  try {
    const response = await fetch(uri)
    const repos = await response.json()
    dispatch(fetchReposSuccess(repos))
  } catch (error) {
    dispatch(fetchReposFailure(error))
  }
}

export const fetchReposRequest = () => ({
  type: types.FETCH_REPOS_REQUEST,
})

export const fetchReposSuccess = data => ({
  type: types.FETCH_REPOS_SUCCESS,
  payload: data,
})

export const fetchReposFailure = error => ({
  type: types.FETCH_REPOS_FAILURE,
  payload: error,
})

export const selectRepo = repo => ({
  type: types.SELECT_REPO,
  payload: repo,
})
