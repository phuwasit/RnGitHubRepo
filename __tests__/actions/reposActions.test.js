import mockStore from '../../config/mock/redux-mock-store'
import * as actions from '../../src/actions/reposActions'

import reposResponseSuccess from '../../__mock__/repos/repos_success.json'
import reposResponseFailure from '../../__mock__/repos/repos_failure.json'

const store = mockStore()

describe('Async repos actions', () => {
  beforeEach(() => {
    store.clearActions()
  })

  it('should response success', async () => {
    fetch.mockResponseSuccess(JSON.stringify(reposResponseSuccess))

    await store.dispatch(actions.fetchReposRequest('facebook', 'react-native'))
    expect(store.getActions()).toMatchSnapshot()
  })

  it('should response failure', async () => {
    fetch.mockResponseFailure(JSON.stringify(reposResponseFailure))

    await store.dispatch(actions.fetchReposRequest('facebook'))
    expect(store.getActions()).toMatchSnapshot()
  })
})
