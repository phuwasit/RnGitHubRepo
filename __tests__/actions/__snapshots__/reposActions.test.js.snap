// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Async repos actions should response failure 1`] = `
Array [
  Object {
    "type": "FETCH_REPOS_REQUEST",
  },
]
`;

exports[`Async repos actions should response success 1`] = `
Array [
  Object {
    "type": "FETCH_REPOS_REQUEST",
  },
]
`;
